File: go/problems/homework_at_given_time.go
package problems

// https://leetcode.com/problems/number-of-students-doing-homework-at-a-given-time
func BusyStudent(startTime []int, endTime []int, queryTime int) int {
	result := 0
	for i, v := range startTime {
		if v <= queryTime && queryTime <= endTime[i] {
			result+= 1
		}
	}
	return result
}

File: rust/src/problems/homework_at_given_time.rs
// https://leetcode.com/problems/number-of-students-doing-homework-at-a-given-time/

pub fn busy_student(start_time: Vec<i32>, end_time: Vec<i32>, query_time: i32) -> i32 {
  let mut result: i32 = 0;
  for (i, start) in start_time.iter().enumerate() {
    if end_time[i] > query_time && *start < query_time {
      result += 1
    }
  }
  return result;
}
